### PR TITLE
MLDB-1599 add clamp function

### PR DESCRIPTION
**Pull request by steadtler**

### CODE REVIEW

File: sql/builtin_functions.cc [hunk 1]
```diff
@@ -2432,6 +2433,50 @@ BoundFunction flatten(const std::vector<BoundSqlExpression> & args)
...
+    return {[=] (const std::vector<ExpressionValue> & args,
+                 const SqlRowScope & scope) -> ExpressionValue
+            {
+                double lower = args[1].toDouble();
+                double upper = args[2].toDouble();
+
+                Date limitsTs = std::max(args[1].getEffectiveTimestamp(), args[2].getEffectiveTimestamp());
+
+                if (args[0].isAtom()) {
+                    double clamped = boost::algorithm::clamp(args[0].toDouble(), lower, upper);
```
Comment (jeremybarnes) [inline sql/builtin_functions.cc]: There is a more pervasive problem... in that 64 bit integer values will end up being molested by being converted to doubles.  Instead, I would define clamp using operator < on the ExpressionValue objects and ignore doubles altogether.

File: sql/builtin_functions.cc [hunk 2]
```diff
@@ -2432,6 +2433,50 @@ BoundFunction flatten(const std::vector<BoundSqlExpression> & args)
 
 static RegisterBuiltin registerFlatten(flatten, "flatten");
 
+BoundFunction clamp(const std::vector<BoundSqlExpression> & args)
+{
+    // Return the result indexed on a single dimension
```
Comment (jeremybarnes) [inline sql/builtin_functions.cc]: Comment doesn't make sense

File: sql/builtin_functions.cc (hunk 1 above)
Comment (jeremybarnes) [inline sql/builtin_functions.cc]: (or operator < on CellValues)

**Comment by jeremybarnes** [comment]
Missing documentation and the test case seems to fail.  Once issues are addressed I'm +1.

**Comment by steadtler** [comment]
DoD got expanded.

**Pull request closed without merging**